Bottom: 3edb6161e2ade2b2b9935f980c2e3f564b67c96d
Top:    6ae6a684dc7dfe26c59a9b6153eb3e003b08efa9
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-24 14:38:47 -0400

Refresh of stac9766-codec

---

diff --git a/sound/soc/codecs/stac9766.c b/sound/soc/codecs/stac9766.c
index 1eff53c..9d60f19 100644
--- a/sound/soc/codecs/stac9766.c
+++ b/sound/soc/codecs/stac9766.c
@@ -276,14 +276,22 @@ static int ac97_analog_prepare(struct snd_pcm_substream *substream,
 {
 	struct snd_soc_codec *codec = dai->codec;
 	struct snd_pcm_runtime *runtime = substream->runtime;
-	u16 vra;
+	unsigned short reg, vra;
 
 	printk("stac9766: ac97_analog_prepare rate %d\n", runtime->rate);
 
 	vra = stac9766_ac97_read(codec, AC97_EXTENDED_STATUS);
 	stac9766_ac97_write(codec, AC97_EXTENDED_STATUS, vra | 0x1);
 
-	return stac9766_ac97_write(codec, AC97_PCM_FRONT_DAC_RATE, runtime->rate);
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		reg = AC97_PCM_FRONT_DAC_RATE;
+	else
+		reg = AC97_PCM_LR_ADC_RATE;
+
+	vra = stac9766_ac97_read(codec, reg);
+	printk("rate is %x %x\n", vra, runtime->rate/2);
+
+	return stac9766_ac97_write(codec, reg, runtime->rate/2);
 }
 
 static int ac97_digital_prepare(struct snd_pcm_substream *substream,
